frontend/packages/dev-console/src/components/taskruns/__tests__/TaskRunDetails.spec.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { SectionHeading, ResourceSummary } from '@console/internal/components/utils';
+import { Status } from '@console/shared';
+import TaskRunDetailsErrorLog from '../TaskRunDetailsErrorLog';
+import { shallow } from 'enzyme';
+import TaskRunDetails from '../TaskRunDetails';
+
+type Component = typeof TaskRunDetails;
+type Props = React.ComponentProps<Component>;
+const TaskDetailsProps: Props = {
+  obj: {
+    kind: 'TaskRun',
+    metadata: { name: 'abhi', namespace: 'abhi1' },
+    status: {
+      completionTime: '2019-10-29T11:57:53Z',
+      conditions: [
+        {
+          lastTransitionTime: '2019-10-29T11:57:53Z',
+          reason: 'Failed',
+          status: 'True',
+          type: 'Failed',
+        },
+      ],
+    },
+  },
+};
+
+describe('TaskRunDetailsPage', () => {
+  const pipelineRunWrapper = shallow(<TaskRunDetails {...TaskDetailsProps} />);
+  it('Renders a SectionHeading', () => {
+    expect(pipelineRunWrapper.find(SectionHeading).exists());
+  });
+  it('Renders a Resource Summary', () => {
+    expect(pipelineRunWrapper.find(ResourceSummary).exists());
+  });
+  it('Renders a Status', () => {
+    expect(pipelineRunWrapper.find(Status).exists());
+  });
+  it('Renders a Task Run Log snippet', () => {
+    expect(pipelineRunWrapper.find(TaskRunDetailsErrorLog).exists());
+  });
+});
```

Comment:
Testing against static content won't save us from anything. The best it can do is fail a test when we change the layout... there isn't any conditional here that we would be concerned about. Ie, no logic breaks can cause these tests to fail.

Reply:
This is just based on current acceptance criteria. Just to make sure, the code adheres to the AC

Reply:
Unit tests guard against accidental breaks in logic.

Eg. A unit test for `add5()` would be `add5(5).toBe(10)`... you could test positive numbers, negative numbers, strings (should return NaN), objects, etc, etc...

If you wrap your `add5()`:

```
function getOffsetNumber() {
  return add5(37);
}
```
This function doesn't get you much but it allows the caller not to worry about providing this static `37` (this is an overly simple example). A testing of this function doesn't mean much. `getOffsetNumber().toBe(42)`... how will this test break? If we choose `37` is not a valid constant anymore? So we just change the test and the code at the same time and nothing super shocking here happens. No assistance is really provided by our test, no assurances we don't accidentally break.

Tests are so you can go into `add5` and change the fact that it doesn't add 5 anymore, it subtracts 8 or does something else (uses math.random for instance). Zero inputs or inputs that don't change the output are not test worthy scenarios. They just explicitly tie our tests to our implementation and thus makes us update one more place when we need to change something. It doesn't guard against breaks.

With this mental model, React Components have a similar structure with 1 extra condition. We only shallowly look at the structure of a component. So if your React Component does not take any props or the props are not conditional rendering props but just pass-thru props, then there are no tests needing to be written. 

So in summary, because I ranted a bit here. Don't test a component that does not use conditional rendering as you're not really writing a test, you're writing just 1 more spot to update if we need to change anything.